python/hail/linalg/blockmatrix.py

```diff
@@ -914,6 +914,15 @@ def sparsify_row_intervals(self, starts, stops, blocks_only=False):
         :class:`.BlockMatrix`
             Sparse block matrix.
         """
+        if isinstance(starts, np.ndarray):
+            if not (starts.dtype==np.int32 or starts.dtype==np.int64):
+                raise ValueError(f"sparsify_row_intervals: starts ndarray must have dtype int32 or int64")
+            starts = [int(s) for s in starts]
+        if isinstance(stops, np.ndarray):
```

Comment:
I think you should add a type checker that expects a integer ndarray or numeric ndarray depending on the application. Then this checking code will just be done automatically.